CRM/Core/Payment.php

```diff
@@ -599,7 +599,7 @@ public function getText($context, $params) {
           return ts('To complete this transaction, click the <strong>Continue</strong> button below.');
         }
         if ($this->_paymentProcessor['billing_mode'] == 4) {
-          return ts('Click the <strong>Continue</strong> button to go to %1, where you will select your payment method and complete the contribution.', [$this->_paymentProcessor['payment_processor_type']]);
+          return ts('Click the <strong>Continue</strong> button to complete the contribution.');
         }
         if ($params['is_payment_to_existing']) {
           return ts('To complete this transaction, click the <strong>Make Payment</strong> button below.');
```

Comment:
And then below this we have a catch-all return text, which again is essentially the same. So just have a single catch-all return text, remove all other logic.

Makes things much simpler.